src/synopsys-action/tools-parameter.ts

```diff
@@ -243,14 +244,20 @@ export class SynopsysToolsParameter {
           owner: {
             name: githubRepoOwner
           },
-          pull: {
-            number: parseInt(githubPrNumber)
-          },
+          pull: {},
           branch: {
             name: githubBranchName
           }
         }
       }
+
+      if (githubPrNumber != null) {
+        githubData.repository.pull.number = Number(githubPrNumber)
+        if (isCommentFlow && isNaN(githubData.repository.pull.number)) {
+          throw new Error('Coverity/Blackduck automation PR comment can be run only by raising PR/MR')
```

Comment:
Below code should be outside and before if(githubPrNumber!=null).
and Also we need to check for null.
if (isCommentFlow && isNaN(githubData.repository.pull.number)) {
          throw new Error('Coverity/Blackduck automation PR comment can be run only by raising PR/MR')
        }

Reply:
done